src/current_shaper.cpp

```diff
@@ -0,0 +1,127 @@
+#include "current_shaper.h"
+
+//global instance
+CurrentShaperTask shaper;
+
+CurrentShaperTask::CurrentShaperTask() : MicroTasks::Task() {
+	_changed = false;
+	_enabled = false;
+}
+
+CurrentShaperTask::~CurrentShaperTask() {
+	// should be useless but just in case
+	evse.release(EvseClient_OpenEVSE_Shaper);
+}
+
+void CurrentShaperTask::setup() {
+
+}
+
+unsigned long CurrentShaperTask::loop(MicroTasks::WakeReason reason) {
+	if (_enabled) {
+			EvseProperties props;
+			if (_changed) {
+				props.setChargeCurrent(_chg_cur);
+				if (_chg_cur < evse.getMinCurrent() ) {
+					// pause temporary, not enough amps available
+					props.setState(EvseState::Disabled);
+				}
+				else {
+					props.setState(EvseState::None);
+				}
+				_changed = false;
+				_timer = millis();
+				evse.claim(EvseClient_OpenEVSE_Shaper,EvseManager_Priority_Limit, props);
```

Comment:
I think maybe we should be using a new priority for safety features at say `5000`, Limit is still a user choice and we will be using it in the future for the limiting features (like time and energy limits) so don't want those to be able to override the current shaper

```suggestion
				evse.claim(EvseClient_OpenEVSE_Shaper,EvseManager_Priority_Safety, props);
```

Reply:
> I think maybe we should be using a new priority for safety features at say `5000`, Limit is still a user choice and we will be using it in the future for the limiting features (like time and energy limits) so don't want those to be able to override the current shaper

Yeah good point, I've changed that.